app/controllers/api/v1/application_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Api::V1::ApplicationController < ActionController::Base
+  include CurrentUser
+  
+  def login_user
+    warden.authenticate(:basic) || unauthorized!
```

Comment:
yes!

maybe even CurrentUser looks for `ALLOWED_AUTH_METHODS` and application_controller and api controller both set them ... no defaults needed ... but must access them via `self.class.ALLOWED_AUTH_METHODS` for it to work